http2_general/test_h2_stream_priority.py

```diff
@@ -0,0 +1,758 @@
+            )
+            client.make_request(
+                self.post_request,
+                end_stream=True,
+                priority_weight=257 - weight,
+                priority_depends_on=stream_id,
+                priority_exclusive=False,
+            )
+            weight = weight + 1
+
+        self.wait_for_responses(client, timeout=240)
+        self.check_response_sequence(client, 256 + 256 * 2)
+
+
+class TestMaxConcurrentStreams(TestPriorityBase, NetWorker):
+    tempesta = {
+        "config": """
+            listen 443 proto=h2;
+            max_concurrent_streams 10;
+            srv_group default {
+                server ${server_ip}:8000;
+            }
+            vhost good {
+                proxy_pass default;
+            }
+            tls_certificate ${tempesta_workdir}/tempesta.crt;
+            tls_certificate_key ${tempesta_workdir}/tempesta.key;
+            tls_match_any_server_name;
+            http_max_header_list_size 0;
+            frang_limits {
+                http_hdr_len 0;
+                http_header_cnt 0;
+            }
+
+            block_action attack reply;
+            block_action error reply;
+            http_chain {
+                host == "bad.com"   -> block;
+                                    -> good;
+            }
+        """
+    }
+
+    def test_max_concurent_stream_exceed_by_stream(self):
+        """
+        If creation of new stream leads to exceeding of max_concurrent_streams, we should reset
+        this stream. Check that all streams, which creation leads to exceedion of max_concurrent_streams
+        will be reset and all previos streams will be finished successfully.
+        """
+        client, server = self.setup_test_priority()
+
+        prev_stream_id = 0
+        self.assertTrue(client.h2_connection.remote_settings.max_concurrent_streams == 10)
+
+        # Create streams with "idle" state with the id range from 21 to 39
+        for i in range(0, client.h2_connection.remote_settings.max_concurrent_streams):
+            stream_id = 2 * client.h2_connection.remote_settings.max_concurrent_streams + 2 * i + 1
+            client.send_bytes(
+                PriorityFrame(
+                    stream_id=stream_id,
+                    depends_on=prev_stream_id,
+                    stream_weight=1,
+                    exclusive=False,
+                ).serialize()
+            )
+            prev_stream_id = stream_id
+
+        valid_req_num = client.valid_req_num
+        """
+        Try to open streams with the id range from 1 to 19, they should be reseted,
+        because of exceed of max concurent stream limit.
+        """
+        for i in range(0, client.h2_connection.remote_settings.max_concurrent_streams):
+            stream_id = client.stream_id
+            client.make_request(self.post_request)
+            self.assertTrue(client.wait_for_reset_stream(stream_id=stream_id))
+
+        return client, valid_req_num
```

Comment:
I always expect 0 for `valid_req_num` here. I think you should remove it

Reply:
Yes you right, fixed